server/chat-plugins/helptickets.js

```diff
@@ -235,13 +274,65 @@ class HelpTicket extends Rooms.RoomGame {
 			const user = Users(ticketGameUser.userid);
 			if (user) user.updateSearch();
 		}
+		if (!this.involvedStaff.length) {
+			if (staff.isStaff && staff.userid !== this.ticket.userid) {
+				this.involvedStaff.push(staff.userid);
+			} else {
+				this.involvedStaff.push(toID(this.ticket.claimed));
+			}
+		}
+		this.writeStats(result);
+	}
+
+	/**
+	 * @param {(boolean | 'ticketban' | 'deleted')} result
+	 */
+	writeStats(result) {
+		// Only run when a ticket is closed/banned/deleted
+		this.closeTime = Date.now();
+		if (this.lastUnclaimedStart) this.unclaimedTime += this.closeTime - this.lastUnclaimedStart;
+		if (!this.ticket.active) {
+			this.resolution = "dead";
+		} else if (!this.firstClaimTime) {
+			this.resolution = "unresolved";
+		} else {
+			this.resolution = "resolved";
+		}
+		if (typeof result === 'boolean') {
+			switch (this.ticket.type) {
+			case 'Appeal':
+			case 'IP-Appeal':
+			case 'ISP-Appeal':
+				this.result = (result ? 'approved' : 'denied');
+				break;
+			case 'PM Harassment':
+			case 'Battle Harassment':
+			case 'Inappropriate Username':
+			case 'Inappropriate Pokemon Nicknames':
+				this.result = (result ? 'valid' : 'invalid');
+				break;
+			case 'Public Room Assistance Request':
+			case 'Other':
+			default:
+				this.result = (result ? 'assisted' : 'unassisted');
+				break;
+			}
+		} else {
+			this.result = result;
+		}
+		let firstClaimWait = 0;
+		if (this.activationTime) firstClaimWait = (this.firstClaimTime ? this.firstClaimTime : this.closeTime) - this.activationTime;
+		// Write to TSV
+		// ticketType\ttotalTime\ttimeToFirstClaim\tinactiveTime\tresolution\tresult\tstaff,userids,seperated,with,commas
+		let line = `${this.ticket.type}\t${(this.closeTime - this.createTime)}\t${firstClaimWait}\t${this.unclaimedTime}\t${this.resolution}\t${this.result}\t${this.involvedStaff.join(',')}`;
```

Comment:
Nitpick.
```suggestion
		const line = `${this.ticket.type}\t${(this.closeTime - this.createTime)}\t${firstClaimWait}\t${this.unclaimedTime}\t${this.resolution}\t${this.result}\t${this.involvedStaff.join(',')}`;
```

Reply:
(There's a lot more of these actually, these will all be fixed when this is TypeScript-ed anyway, so it's whatever I guess.)